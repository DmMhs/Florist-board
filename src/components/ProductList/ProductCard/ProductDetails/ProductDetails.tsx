import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';

import './ProductDetails.less';

class ProductDetails extends Component {
  render() {
    const params = JSON.parse(
      decodeURIComponent((this.props as any).match.params.data)
    );
    const { title, price, images, currency } = params;

    let newUrl = '';
    const firstPart = images[0].slice(0, 72);
    let secondPart = images[0].slice(72, images[0].length);
    secondPart = secondPart.replace(/\//gi, '%2F');
    newUrl = firstPart + secondPart;

    const imgStyle = {
      background: `url(${newUrl})`,
      backgroundPosition: 'center',
      backgroundSize: 'cover',
      backgroundRepeat: 'no-repeat'
    };
    return (
      <div className="ProductDetails">
        <h1>{title.toUpperCase()}</h1>
        <div className="product-info-wrapper">
          <div className="image-wrapper">
            <div className="image" style={imgStyle} />
            <h3 className="price">
              only <span className="accent">{price}$</span>
            </h3>
            <button className="shopping-btn" type="button">
              <NavLink to="/shop">GO SHOPPING</NavLink>
            </button>
          </div>

          <div className="info">
            <h2>Description</h2>
            <hr />
            <p>
              Let's make some happy little clouds in our world. Here's something
              that's fun. See there, told you that would be easy. Maybe he has a
              little friend that lives right over here. Those great big fluffy
              clouds. Every day I learn. Isn't that fantastic? You can just push
              a little tree out of your brush like that. You don't want to kill
              all your dark areas they are very important. Imagination is the
              key to painting. This is the time to get out all your
              flustrations, much better than kicking the dog around the house or
              taking it out on your spouse. We can always carry this a step
              further. There's really no end to this.
            </p>
          </div>
        </div>
      </div>
    );
  }
}

export default ProductDetails;
