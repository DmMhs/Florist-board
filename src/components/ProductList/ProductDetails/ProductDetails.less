.ProductDetails {
  margin-bottom: 2rem;
  h1 {
    font-weight: lighter;
  }
  .product-info-wrapper {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: flex-start;
    .image-wrapper {
      .image {
        height: 280px;
        width: 280px;
        margin: 0 20px;
        border-radius: 7px;
        box-shadow: 1px 1px 5px #ccc;
        background-position: center;
        background-size: cover;
        background-repeat: no-repeat;
      }
      .price {
        font-size: 2rem;
        font-weight: lighter;
      }

      .button {
        cursor: pointer;
        margin: 0.5rem auto;
        width: fit-content;
        display: flex;
        flex-direction: row;
        align-items: center;
        justify-content: space-around;
        background-color: white;
        border: 1.5px solid rgb(59, 89, 152);
        box-sizing: border-box;
        flex-wrap: nowrap;
        transition: 0.5s;
        span {
          color: rgb(59, 89, 152);
          text-align: center;
          padding: 5px;
        }
        &:hover {
          transition: 0.5s;
          background-color: rgb(59, 89, 152);
          span {
            color: white;
          }
        }
      }

      .shopping-btn {
        background-color: rgb(144, 201, 144);
        color: white;
        border: none;
        padding: 7px;
        border-radius: 5px;
        font-size: 1.3rem;
        cursor: pointer;
        &:hover {
          background-color: rgb(105, 211, 105);
        }
        a {
          text-decoration: none;
          color: white;
          &:hover,
          &:active,
          &:visited {
            color: white;
          }
        }
      }
    }

    .info {
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;
      flex-basis: 50%;
      padding: 20px;
      color: #333333;
      h2 {
        width: 50%;
        font-size: 2rem;
        font-weight: lighter;
      }
      hr {
        width: 50%;
        border: 0.5px solid #333333;
      }
      p {
        font-size: 1.2rem;
        text-align: justify;
        padding: 0;
      }
    }
  }
  .accent {
    color: rgb(105, 211, 105);
  }
}
@media screen and (max-width: 768px) {
  .ProductDetails {
    .product-info-wrapper {
      display: flex;
      flex-direction: column;
      justify-content: flex-start;
      align-items: center;

      .info {
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        flex-basis: 100%;
      }
    }
  }
}
