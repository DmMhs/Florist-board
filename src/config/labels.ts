const labels = {
  brand: 'FLORIST.UA',
  adress: 'MARKET SQUARE, LVIV, UA',
  postCode: '346346',
  phone: '+123-45-678',
  navigation: {
    home: 'HOME',
    shop: 'SHOP',
    gallery: 'GALLERY',
    contacts: 'CONTACTS',
    cart: 'CART',
    account: {
      main: 'MY ACCOUNT',
      menu: {
        signIn: 'SIGN IN',
        signUp: 'SIGN UP'
      }
    }
  }
};

export default labels;
