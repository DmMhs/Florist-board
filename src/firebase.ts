import * as firebase from 'firebase';

export const config = {
  apiKey: 'AIzaSyB7esiw1XYzS2_hllELqqYzqN5wIQav0Oc',
  authDomain: 'florist-cb933.firebaseapp.com',
  databaseURL: 'https://florist-cb933.firebaseio.com',
  projectId: 'florist-cb933',
  storageBucket: 'florist-cb933.appspot.com',
  messagingSenderId: '582977319235'
};

firebase.initializeApp(config);
export const database = firebase.database();
export const homeImagesRef = database.ref('home-images');
export const productsRef = database.ref('products');
export const galleryImagesRef = database.ref('gallery-images');
