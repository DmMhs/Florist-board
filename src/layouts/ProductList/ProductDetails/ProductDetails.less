.ProductDetails {
  min-height: calc(100vh - 7rem);
  border: 1px solid transparent;
  box-sizing: border-box;
  h1 {
    font-weight: lighter;
  }
  .product-info-wrapper {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: flex-start;
    .image-wrapper {
      .image {
        height: 280px;
        width: 280px;
        margin: 0 20px;
        border-radius: 7px;
        box-shadow: 1px 1px 5px #ccc;
        background-position: center;
        background-size: cover;
        background-repeat: no-repeat;
        box-sizing: border-box;
      }
      .price {
        font-size: 2rem;
        font-weight: lighter;
      }

      .button {
        cursor: pointer;
        margin: 0.5rem auto;
        width: fit-content;
        display: flex;
        flex-direction: row;
        align-items: center;
        justify-content: space-around;
        background-color: white;
        border: 1.5px solid rgb(59, 89, 152);
        box-sizing: border-box;
        flex-wrap: nowrap;
        transition: 0.5s;
        span {
          color: rgb(59, 89, 152);
          text-align: center;
          padding: 5px;
        }
        &:hover {
          transition: 0.5s;
          background-color: rgb(59, 89, 152);
          span {
            color: white;
          }
        }
      }

      .shopping-btn {
        box-sizing: border-box;
        padding: 7px;
        font-size: 1.3rem;
        cursor: pointer;
        border: 1.5px solid rgb(120, 192, 120);
        background-color: white;
        transition: 0.5s;
        &:hover {
          background-color: rgb(120, 192, 120);
          transition: 0.5s;
          a {
            span {
              color: white;
            }
            i {
              color: white;
            }
          }
        }
        a {
          text-decoration: none;
          span {
            display: inline;
            font-size: 1.2rem;
            color: rgb(120, 192, 120);
            &:hover,
            &:active,
            &:visited {
              color: white;
            }
          }
          i {
            color: rgb(120, 192, 120);
          }
        }
      }
    }

    .info {
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;
      flex-basis: 50%;
      padding: 20px;
      color: #333333;
      h2 {
        width: 50%;
        font-size: 2rem;
        font-weight: lighter;
      }
      hr {
        width: 50%;
        border: 0.5px solid #333333;
      }
      p {
        font-size: 1.2rem;
        text-align: justify;
        padding: 0;
      }
    }
  }
  .accent {
    color: rgb(105, 211, 105);
  }
}
@media screen and (max-width: 768px) {
  .ProductDetails {
    margin-top: 4rem;
    .product-info-wrapper {
      display: flex;
      flex-direction: column;
      justify-content: flex-start;
      align-items: center;

      .info {
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        flex-basis: 100%;
      }
    }
  }
}
