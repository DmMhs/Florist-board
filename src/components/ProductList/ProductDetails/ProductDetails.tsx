import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { FacebookShareButton, FacebookIcon } from 'react-share';

import Spinner from '../../Spinner/Spinner';
import { Product } from '../../../models/Product';
import { productsRef } from '../../../firebase';
import './ProductDetails.less';
import Helmet from 'react-helmet';
import { AuthContext } from '../../Auth/AuthContext';

interface MatchParams {
  id: string;
}

interface Props extends RouteComponentProps<MatchParams> {}

interface RouteComponentProps<P> {
  match: match<P>;
}

interface match<P> {
  params: P;
  isExact: boolean;
  path: string;
  url: string;
}
interface ProductDetailsState {
  productData: Product;
  fetchInProgress: boolean;
}

class ProductDetails extends Component<
  RouteComponentProps<MatchParams>,
  ProductDetailsState
> {
  constructor(props: RouteComponentProps<MatchParams>) {
    super(props);
    this.state = {
      productData: {
        title: '',
        images: [],
        price: 0,
        currency: '',
        available: false,
        description: ''
      },
      fetchInProgress: false
    };
  }
  componentDidMount() {
    this.setState({
      fetchInProgress: true
    });
    productsRef
      .child(this.props.match.params.id)
      .once('value')
      .then(snapshot => {
        this.setState({
          productData: snapshot.val(),
          fetchInProgress: false
        });
      })
      .catch(error => console.log(error));
  }
  render() {
    const imgStyle = {
      backgroundImage: `url(${this.state.productData.images[0]})`
    };
    return (
      <AuthContext.Consumer>
        {value =>
          value && (
            <div className="ProductDetails">
              {this.state.fetchInProgress === false ? (
                <div>
                  <h1>{this.state.productData.title.toUpperCase()}</h1>
                  <div className="product-info-wrapper">
                    <div className="image-wrapper">
                      <div className="image" style={imgStyle} />
                      <h3 className="price">
                        {this.state.productData.available === false ? (
                          <span>
                            not in stock <i className="far fa-frown" />
                          </span>
                        ) : (
                          <div>
                            only{' '}
                            <span className="accent">
                              {this.state.productData.price}$
                            </span>
                          </div>
                        )}
                      </h3>
                      {value.state.authenticationMethod === 'facebook' ? (
                        <div className="button-wrapper">
                          <FacebookShareButton
                            url={`https://florist-ua.herokuapp.com/product-details/${
                              this.props.match.params.id
                            }`}
                          >
                            <div>
                              <span> SHARE</span>
                              <FacebookIcon size={50} />
                            </div>
                          </FacebookShareButton>
                        </div>
                      ) : null}

                      <button className="shopping-btn" type="button">
                        <NavLink to="/shop">
                          GO SHOPPING <i className="fas fa-shopping-cart" />
                        </NavLink>
                      </button>
                    </div>

                    <div className="info">
                      <h2>Description</h2>
                      <hr />
                      <p>{this.state.productData.description}</p>
                      <img
                        src={this.state.productData.images[1]}
                        style={{ maxHeight: '200px' }}
                      />
                    </div>
                  </div>
                </div>
              ) : (
                <Spinner />
              )}
            </div>
          )
        }
      </AuthContext.Consumer>
    );
  }
}

export default ProductDetails;
