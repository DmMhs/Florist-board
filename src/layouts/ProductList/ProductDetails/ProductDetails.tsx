import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';

import {Spinner} from '../../../components';
import { Product } from '../../../models/Product';
import { productsRef } from '../../../firebase';
import { AppContext } from '../../../AppContext';
import { BASE_URL } from '../../../config/main';
import { shareOverrideOGMeta } from '../../../services/share'

import './ProductDetails.less';

interface MatchParams {
  id: string;
}

interface Props extends RouteComponentProps<MatchParams> {}

interface RouteComponentProps<P> {
  match: Match<P>;
}

interface Match<P> {
  params: P;
  isExact: boolean;
  path: string;
  url: string;
}
interface ProductDetailsState {
  productData: Product;
  fetchInProgress: boolean;
}

class ProductDetails extends Component<
  RouteComponentProps<MatchParams>,
  ProductDetailsState
> {
  constructor(props: RouteComponentProps<MatchParams>) {
    super(props);
    this.state = {
      productData: {
        title: '',
        title_uk: '',
        images: [],
        price: 0,
        currency: '',
        available: false,
        description: '',
        description_uk: ''
      },
      fetchInProgress: false
    };
  }

  public componentDidMount() {
    this.setState({
      fetchInProgress: true
    });
    productsRef
      .child(this.props.match.params.id)
      .once('value')
      .then(snapshot => {
        this.setState({
          productData: snapshot.val(),
          fetchInProgress: false
        });
      })
      .catch(error => console.log(error));
  }

  public render() {
    const {
      title,
      title_uk,
      description,
      description_uk,
      available,
      price,
      images
    } = this.state.productData;

    const imgStyle = {
      backgroundImage: `url(${this.state.productData.images[0]})`
    };

    const context = this.context;
    const labels = context.state.labels;
    const lang = context.state.lang;
    const labelsRoot =
      labels[lang].pages.productDetails;

    return (
      <AppContext.Consumer>
        {value => (
          <div className="ProductDetails">
            {this.state.fetchInProgress === false ? (
              <div>
                <h1>
                  {lang === 'en'
                    ? title.toUpperCase()
                    : title_uk.toUpperCase()}
                </h1>
                <div className="product-info-wrapper">
                  <div className="image-wrapper">
                    <div className="image" style={imgStyle} />
                    <h3 className="price">
                      {available === false ? (
                        <span>
                          {labelsRoot.notAvailable}{' '}
                          <i className="far fa-frown" />
                        </span>
                      ) : (
                        <div>
                          {labelsRoot.only}{' '}
                          <span className="accent">{price}$</span>
                        </div>
                      )}
                    </h3>
                    {value.state.authenticationMethod === 'facebook' ? (
                      <div
                        className="button"
                        onClick={shareOverrideOGMeta.bind(
                          this,
                          BASE_URL +
                            `/product-details/${this.props.match.params.id}`,
                          value.state.lang === 'en' ? title : title_uk,
                          value.state.lang === 'en'
                            ? (description as string)
                            : (description_uk as string),
                          images[0]
                        )}
                      >
                        <span>
                          {labelsRoot.facebookShare}{' '}
                          <i className="fab fa-facebook-f" />
                        </span>
                      </div>
                    ) : null}

                    <button className="shopping-btn" type="button">
                      <NavLink to="/shop">
                        <span>{labelsRoot.goShopping}</span>{' '}
                        <i className="fas fa-shopping-cart" />
                      </NavLink>
                    </button>
                  </div>

                  <div className="info">
                    <h2>{labelsRoot.description}</h2>
                    <hr />
                    <p>
                      {lang === 'en' ? description : description_uk}
                    </p>
                  </div>
                </div>
              </div>
            ) : (
              <Spinner />
            )}
          </div>
        )}
      </AppContext.Consumer>
    );
  }
}

ProductDetails.contextType = AppContext;

export default ProductDetails;
