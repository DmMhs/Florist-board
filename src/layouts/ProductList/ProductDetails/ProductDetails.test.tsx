import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';

import ProductDetails from './ProductDetails';
import { productsRef } from '../../../firebase';
import AppContextProvider from '../../../AppContext';
import labels from '../../../config/labels';

describe('ProductDetails works as expected', () => {
  it('renders without crashing', () => {
    const wrapper = mount(
      <BrowserRouter>
        <AppContextProvider>
          <ProductDetails
            match={{
              params: { id: '-Laz5nc-kJiI1uDubC9K' },
              isExact: true,
              path: '',
              url: ''
            }}
          />
        </AppContextProvider>
      </BrowserRouter>
    );
    const context = wrapper.find('AppContextProvider').instance();
    context.setState({
      lang: 'en',
      labels: labels,
      fetchInProgress: false,
      mobileMode: true,
      showNavigation: false,
      togglePosition: 'absolute'
    });
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('details must be rendered', () => {
    const wrapper = mount(
      <BrowserRouter>
        <AppContextProvider>
          <ProductDetails
            match={{
              params: { id: '-Laz5nc-kJiI1uDubC9K' },
              isExact: true,
              path: '',
              url: ''
            }}
          />
        </AppContextProvider>
      </BrowserRouter>
    );
    const context = wrapper.find('AppContextProvider').instance();
    context.setState({
      lang: 'en',
      labels: labels,
      fetchInProgress: false,
      mobileMode: true,
      showNavigation: false,
      togglePosition: 'absolute'
    });
    wrapper.update();
    expect(wrapper.find('.ProductDetails').exists()).toBeTruthy();
  });

  it('product data do fetch', async () => {
    const wrapper = mount(
      <BrowserRouter>
        <AppContextProvider>
          <ProductDetails
            match={{
              params: { id: '-Laz5nc-kJiI1uDubC9K' },
              isExact: true,
              path: '',
              url: ''
            }}
          />
        </AppContextProvider>
      </BrowserRouter>
    );
    const context = wrapper.find('AppContextProvider').instance();
    context.setState({
      lang: 'en',
      labels: labels,
      fetchInProgress: false,
      mobileMode: true,
      showNavigation: false,
      togglePosition: 'absolute'
    });
    wrapper.update();
    const instance = wrapper.find('ProductDetails').instance();
    instance.state = {
      productData: {},
      fetchInProgress: false
    };
    await productsRef
      .child(instance.props.match.params.id)
      .once('value')
      .then(snapshot => {
        instance.setState({
          productData: snapshot.val(),
          fetchInProgress: false
        });
      });
    expect(instance.state.productData.images.length).toBeGreaterThan(0);
  });
  it('Details are displayed when fetch is over', () => {
    const wrapper = mount(
      <BrowserRouter>
        <AppContextProvider>
          <ProductDetails
            match={{
              params: { id: 'any' },
              isExact: true,
              path: '',
              url: ''
            }}
          />
        </AppContextProvider>
      </BrowserRouter>
    );
    const context = wrapper.find('AppContextProvider').instance();
    context.setState({
      lang: 'en',
      labels: labels,
      fetchInProgress: false,
      mobileMode: true,
      showNavigation: false,
      togglePosition: 'absolute'
    });
    wrapper.update();
    const instance = wrapper.find('ProductDetails').instance();
    instance.setState({
      fetchInProgress: true
    });
    expect(wrapper.find('.product-info-wrapper').exists()).toBeFalsy();
    instance.setState({
      fetchInProgress: false
    });
    wrapper.update();
    expect(wrapper.find('.product-info-wrapper').exists()).toBeTruthy();
  });
});
