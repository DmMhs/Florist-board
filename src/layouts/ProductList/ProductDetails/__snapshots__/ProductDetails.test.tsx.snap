// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetails works as expected renders without crashing 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(AppContextProvider)>
      <Route>
        <AppContextProvider
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <ProductDetails
            match={
              Object {
                "isExact": true,
                "params": Object {
                  "id": "-Laz5nc-kJiI1uDubC9K",
                },
                "path": "",
                "url": "",
              }
            }
          >
            <div
              className="ProductDetails"
            >
              <WithStyles(CircularIndeterminate)>
                <CircularIndeterminate
                  classes={
                    Object {
                      "progress": "CircularIndeterminate-progress-12",
                    }
                  }
                >
                  <div
                    className="Spinner"
                  >
                    <WithStyles(CircularProgress)
                      className="CircularIndeterminate-progress-12"
                    >
                      <CircularProgress
                        className="CircularIndeterminate-progress-12"
                        classes={
                          Object {
                            "circle": "MuiCircularProgress-circle-19",
                            "circleDisableShrink": "MuiCircularProgress-circleDisableShrink-22",
                            "circleIndeterminate": "MuiCircularProgress-circleIndeterminate-21",
                            "circleStatic": "MuiCircularProgress-circleStatic-20",
                            "colorPrimary": "MuiCircularProgress-colorPrimary-16",
                            "colorSecondary": "MuiCircularProgress-colorSecondary-17",
                            "indeterminate": "MuiCircularProgress-indeterminate-15",
                            "root": "MuiCircularProgress-root-13",
                            "static": "MuiCircularProgress-static-14",
                            "svg": "MuiCircularProgress-svg-18",
                          }
                        }
                        color="primary"
                        disableShrink={false}
                        size={40}
                        thickness={3.6}
                        value={0}
                        variant="indeterminate"
                      >
                        <div
                          className="MuiCircularProgress-root-13 MuiCircularProgress-colorPrimary-16 MuiCircularProgress-indeterminate-15 CircularIndeterminate-progress-12"
                          role="progressbar"
                          style={
                            Object {
                              "height": 40,
                              "width": 40,
                            }
                          }
                        >
                          <svg
                            className="MuiCircularProgress-svg-18"
                            viewBox="22 22 44 44"
                          >
                            <circle
                              className="MuiCircularProgress-circle-19 MuiCircularProgress-circleIndeterminate-21"
                              cx={44}
                              cy={44}
                              fill="none"
                              r={20.2}
                              strokeWidth={3.6}
                              style={Object {}}
                            />
                          </svg>
                        </div>
                      </CircularProgress>
                    </WithStyles(CircularProgress)>
                  </div>
                </CircularIndeterminate>
              </WithStyles(CircularIndeterminate)>
            </div>
          </ProductDetails>
        </AppContextProvider>
      </Route>
    </withRouter(AppContextProvider)>
  </Router>
</BrowserRouter>
`;
